src/rust/lib/graph-editor/src/component/node/port/output.rs

```diff
@@ -0,0 +1,365 @@
+
+            eval mouse_over_while_inactive ([delay_show,delay_hide](_){
+                delay_hide.stop();
+                delay_show.rewind();
+                delay_show.set_end_value(TWEEN_END_VALUE);
+            });
+            eval port_mouse_out ([delay_hide,delay_show](_){
+                delay_show.stop();
+                delay_hide.rewind();
+                delay_hide.set_end_value(TWEEN_END_VALUE);
+            });
+
+            activate_ports <- any(mouse_over_while_active,on_delay_show_finished);
+            eval_ activate_ports (delay_hide.rewind());
+
+            activate_ports_with_selected <- port_mouse_over.sample(&activate_ports);
+
+            hide_all <- on_delay_hide_finished.map(f_!(delay_show.rewind()));
+
+        }
+
+        // Init ports
+        for (index,view) in data.ports.borrow().iter().enumerate() {
+            let shape        = &view.shape;
+            let port_size    = Animation::<f32>::new(&network);
+            let port_opacity = Animation::<f32>::new(&network);
+
+            frp::extend! { network
+
+
+                // === Mouse Event Handling == ///
+
+                eval_ view.events.mouse_over(port_mouse_over.emit(PortId{index}));
+                eval_ view.events.mouse_out(port_mouse_out.emit(PortId{index}));
+                eval_ view.events.mouse_down(frp.on_port_mouse_down.emit(PortId{index}));
+
+
+                 // === Animation Handling == ///
+
+                 eval port_size.value    ((size) shape.grow.set(*size));
+                 eval port_opacity.value ((size) shape.opacity.set(*size));
+
+
+                // === Visibility and Highlight Handling == ///
+
+                 def _hide_all = hide_all.map(f_!([port_size,port_opacity]{
+                     port_size.set_target_value(0.0);
+                     port_opacity.set_target_value(0.0);
+                 }));
+
+                // Through the provided ID we can infer whether this port should be highlighted.
+                is_selected      <- activate_ports_with_selected.map(move |id| id.index == index);
+                show_normal      <- activate_ports_with_selected.gate_not(&is_selected);
+                show_highlighted <- activate_ports_with_selected.gate(&is_selected);
+
+                eval_ show_highlighted ([port_opacity,port_size]{
+                    port_opacity.set_target_value(1.0);
+                    port_size.set_target_value(HIGHLIGHT_SIZE);
+                });
+
+                eval_ show_normal ([port_opacity,port_size]
+                    port_opacity.set_target_value(0.5);
+                    port_size.set_target_value(BASE_SIZE);
+                );
+            }
+        }
+
+        // FIXME this is a hack to ensure the ports are invisible at startup.
+        // Right now we get some of FRP mouse events on startup that leave the
+        // ports visible by default.
+        // Once that is fixed, remove this line.
+        delay_hide.finish();
```

Comment:
I was asking aobut it in previous PR - tell me more about it. When it happens. It needs to be fixed before those or lands. What events you get and where?